processor/src/main/java/com/tickaroo/tikxml/processor/generator/CodeGeneratorHelper.kt

```diff
@@ -162,57 +226,52 @@ class CodeGeneratorHelper(val customTypeConverterManager: CustomTypeConverterMan
         return builder.build()
     }
 
+
     /**
      * get the assignment statement for reading attributes
      */
-    fun assignViaTypeConverterOrPrimitive(element: Element, assignmentType: AssignmentType, accessResolver: FieldAccessResolver, customTypeConverterQualifiedClassName: String?): CodeBlock {
-
+    fun assignViaTypeConverterOrPrimitive(
+            element: Element,
+            assignmentType: AssignmentType,
+            accessResolver: FieldAccessResolver,
+            customTypeConverterQualifiedClassName: String?
+    ): CodeBlock {
         val type = element.asType()
         val xmlReaderMethodPrefix = assignmentType.xmlReaderMethodPrefix()
+        val codeWriterFormat = "$tikConfigParam.getTypeConverter(%s.class).read($readerParam.$xmlReaderMethodPrefix())"
 
-        val surroundWithTryCatch = fun(assignmentStatement: String) = CodeBlock.builder()
-                .beginControlFlow("try")
-                .add(accessResolver.resolveAssignment(assignmentStatement))
-                .nextControlFlow("catch(\$T e)", ClassName.get(TypeConverterNotFoundException::class.java))
-                .addStatement("throw e")
-                .nextControlFlow("catch(\$T e)", ClassName.get(Exception::class.java))
-                .addStatement("throw new \$T(e)", ClassName.get(IOException::class.java))
-                .endControlFlow()
-                .build()
-
-        val codeWriter = fun(className: String) =
-                "$tikConfigParam.getTypeConverter($className.class).read($readerParam.$xmlReaderMethodPrefix())"
-
-        val codeBlockGen = fun(typesMap: Map<String, String>, resolveMethodName: String) = generatePrimitiveConverter(
-                typesMap,
-                typeConvertersForPrimitives,
-                surroundWithTryCatch,
-                codeWriter
-        ) ?: accessResolver.resolveAssignment("$readerParam.$xmlReaderMethodPrefix$resolveMethodName()")
-
-        return when {
+        var resolveMethodName = ""
```

Comment:
This is a bit strange, but I think we can change that in a future refactoring / cleanup.

I think it would be preferable to have this problem solved in a nicer way like a `sealed class` hierarchy that represents whether or not assignmentState != null and resolvedMethodName is only defined in this sealed class hierarchy if needed.